Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/random_perturbation.h

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2017 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+//
+// Author(s)     : Jane Tournois
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_RANDOM_PERTURBATION_H
+#define CGAL_POLYGON_MESH_PROCESSING_RANDOM_PERTURBATION_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+#include <CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h>
+
+#include <CGAL/Polygon_mesh_processing/internal/named_function_params.h>
+#include <CGAL/Polygon_mesh_processing/internal/named_params_helper.h>
+
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_traits.h>
+#include <CGAL/AABB_face_graph_triangle_primitive.h>
+
+#include <CGAL/Random.h>
+
+#include <boost/foreach.hpp>
+
+#ifdef CGAL_PMP_RANDOM_PERTURBATION_VERBOSE
+#include <CGAL/Timer.h>
+#endif
+
+namespace CGAL {
+
+namespace Polygon_mesh_processing {
+
+namespace internal {
+
+  template<typename GT, typename RNG>
+  typename GT::Vector_3 construct_random_vector_3(const double& max_size,
+                                                  RNG& rng,
+                                                  const GT& gt)
+  {
+    typedef typename GT::FT FT;
+    typename GT::Construct_vector_3 vec = gt.construct_vector_3_object();
+
+    return vec(FT(rng.get_double(-max_size, max_size)),
+               FT(rng.get_double(-max_size, max_size)),
+               FT(rng.get_double(-max_size, max_size)));
+  }
+
+  template<typename GT, typename PM, typename VCMap, typename VPMap, typename RNG>
+  void random_perturbation_impl(PM& tmesh,
+                                const double& max_size,
+                                VCMap vcmap,
+                                VPMap vpmap,
+                                bool do_project,
+                                RNG& rng,
```

Comment:
CGAL::Random() which is passed to random_perturbation_impl is (I think) a rvalue, so can't be referenced, unless considered as a rvalue-reference with the && from C++11. If it must be able to work without C++11, then it's not any better I guess...